test/problems/second_quantization/molecular/test_molecular_problem.py

```diff
@@ -0,0 +1,71 @@
+
+"""Tests Molecular Problem."""
+
+from test import QiskitNatureTestCase
+from qiskit_nature.problems.second_quantization.molecular.fermionic_op_builder import \
+    build_fermionic_op
+from qiskit_nature.transformers import ActiveSpaceTransformer
+from qiskit_nature.drivers import HDF5Driver
+from qiskit_nature.operators.second_quantization import SecondQuantizedOp
+from qiskit_nature.problems.second_quantization.molecular.molecular_problem import MolecularProblem
+
+
+class TestMolecularProblem(QiskitNatureTestCase):
+    """Tests Molecular Problem."""
+
+    def test_second_q_ops_without_transformers(self):
+        """Tests that the list of second quantized operators is created if no transformers
+        provided."""
+        expected_num_of_electr_terms = 185
+        expected_num_of_sec_quant_ops = 7
+
+        driver = HDF5Driver(hdf5_input=self.get_resource_path('H2_631g.hdf5', 'transformers'))
+        transformation = []
+        molecular_problem = MolecularProblem(driver, transformation)
+
+        expected_electr_sec_quant_op = TestMolecularProblem._calc_expected_sec_quant_op(driver,
+                                                                                        None)
+
+        second_quantized_ops = molecular_problem.second_q_ops()
+        electr_sec_quant_op = second_quantized_ops[0]
+        assert electr_sec_quant_op.boson is None
+        assert electr_sec_quant_op.spin == {}
+        assert len(second_quantized_ops) == expected_num_of_sec_quant_ops
+        for second_quantized_op in second_quantized_ops:
+            assert isinstance(second_quantized_op, SecondQuantizedOp)
+        assert len(electr_sec_quant_op.fermion) == expected_num_of_electr_terms
+        assert expected_electr_sec_quant_op.__eq__(electr_sec_quant_op)
+
+    def test_second_q_ops_with_active_space(self):
+        """Tests that the correct second quantized operator is created if an active space
+        transformer is provided."""
+        driver = HDF5Driver(hdf5_input=self.get_resource_path('H2_631g.hdf5', 'transformers'))
+        trafo = ActiveSpaceTransformer(num_electrons=2, num_orbitals=2)
+        expected_electr_sec_quant_op = TestMolecularProblem._calc_expected_sec_quant_op(driver,
+                                                                                        trafo)
+
+        molecular_problem = MolecularProblem(driver, [trafo])
+        second_quantized_ops_list = molecular_problem.second_q_ops()
+        electr_sec_quant_op = second_quantized_ops_list[0]
+
+        assert expected_electr_sec_quant_op.__eq__(electr_sec_quant_op)
+        # TODO test QMolecule itself if it is ever a field in MolecularProblem
+
+    @staticmethod
+    def _calc_expected_sec_quant_op(driver, trafo):
+        q_molecule = driver.run()
+        if trafo:
+            q_molecule = trafo.transform(q_molecule)
+        electronic_fermionic_op = build_fermionic_op(q_molecule)
+        return SecondQuantizedOp([electronic_fermionic_op])
```

Comment:
Yes. It might feel a bit redundant because MolecularProblem does sth similar inside. 

The other option would be copy-pasting the actual content (values) of the `SecondQuantizedOp` that I observe when running the test and assuming it is correct, setting it as an expected behaviour and if any changes in the future will result in a different behaviour, we will see that the test fails, something changed and we might want to assess if we expected such a change. 

I am not sure if I am up to date with all the physics-related knowledge in this particular topic to be able to tell if the observed values actually make sense. Same for building the `FermionicOp` and that's why I was reluctant with asserting actual values. Perhaps I can rely on your feedback with that. Let me know.

Reply:
I have to think about this. I will get back to you :+1:

Reply:
I think I would rather copy-paste the actual contents into the test. It seems less prone to errors in the test suite. @woodsp-ibm @manoelmarques any thoughts on this?

Reply:
Since it returns a list of operators, as it public API, I guess its a question of how to determine that the list is correct. Evidently, as it done the length and the type of each in the list is basic. If we named the operators that could be tested. Aside from storing the terms as a list and checking them, I don't know how else to assure its building the set (list) of operators we expect.  If the operator had some exact diagonalization capability, as mentioned above, that might work too.